uwuno/src/main/java/com/learning/uwuno/services/gameService.java

```diff
@@ -48,34 +49,41 @@ public player addPlayer(String name, String uid) {
         return roomList;
     }
 
-    // Callees must handle NoSuchElementException
     public room getRoom(String uid) {
-        return roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+        return roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst()
+                .orElseThrow(() -> new errorNotFound("Room ID could not be found"));
     }
 
     public player getPlayer(String uid, String pid) {
-        return getRoom(uid).getPlayer(pid);
+        try {
+            return getRoom(uid).getPlayer(pid);
+        }
+        catch (NoSuchElementException e) {
+            throw new errorNotFound("Player ID could not be found");
+        }
     }
 
     // PUTS
     public void updateRoomName(String uid, String newName) {
         if (newName.isBlank())
-            throw new badRequest();
-        room room = roomList.stream().filter(t -> t.getUid().equals(uid)).findFirst().get();
+            throw new badRequest("Room name cannot be blank");
+        room room = getRoom(uid);
```

Comment:
@achang1 Changed to use getRoom() so we don't have to repeat code + error handling is already done